core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3705,21 +3716,32 @@ public void stopSync(AlluxioURI syncPoint)
   /**
    * @param fileId file ID
    * @param jobId persist job ID
+   * @param persistenceWaitTime persistence initial wait time
    * @param uri Alluxio Uri of the file
    * @param tempUfsPath temp UFS path
    */
-  private void addPersistJob(long fileId, long jobId, AlluxioURI uri, String tempUfsPath) {
+  private void addPersistJob(long fileId, long jobId, long persistenceWaitTime, AlluxioURI uri,
+      String tempUfsPath) {
     alluxio.time.ExponentialTimer timer = mPersistRequests.remove(fileId);
     if (timer == null) {
       timer = new alluxio.time.ExponentialTimer(
           ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_INTERVAL_MS),
           ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_INTERVAL_MS),
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_WAIT_TIME_MS),
+          persistenceWaitTime,
```

Comment:
I am under the assumption that `MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS` represents the total amount of time since adding to the job that we will allow before removing/erroring. If that is the case, then the following comment applies:

Similar to another comment I made, is there a case here where a user could specify `persistenceWaitTime` larger than `MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS`? It might be useful to fail-fast in those cases.

Reply:
I add the comment to `USER_FILE_PERSISTENTCE_INITIAL_WAIT_TIME` to say it's not bigger than `TOTAL_WAIT_TIME`. The default total wait time is 1 day, usually it will not be exceeded.